### PR TITLE
JS iframes can't load resources from a domain different than the one from the bootstrap

### PR DESCRIPTION
**Pull request by cyril-sf**
Also:
- Update MessageChannel.js
- Errors inside a sandbox can be propagated to the container.

### CODE REVIEW

**Comment by cyril-sf** [comment]
@hjdivad Let me know what you think

File: lib/oasis/base_adapter.js [hunk 1]
```diff
@@ -85,7 +85,12 @@ BaseAdapter.prototype = {
 
       oasis.configuration.oasisURL = event.data.oasisURL;
 
-      adapter.loadScripts(event.data.base, event.data.scriptURLs, oasis);
+      try {
+        adapter.loadScripts(event.data.base, event.data.scriptURLs, oasis);
+      } catch( e ) {
+        // The required resources can't be loaded, abort
```
Comment (hjdivad) [inline lib/oasis/base_adapter.js]: probably should not suppress errors here

File: lib/oasis/webworker_adapter.js [hunk 2]
```diff
@@ -19,6 +19,15 @@ var WebworkerAdapter = extend(BaseAdapter, {
     worker.name = sandbox.options.url + '?uuid=' + UUID.generate();
     sandbox.worker = worker;
 
+    // Error handling inside the worker
+    worker.errorHandler = function(event) {
+      console.log(event.data);
```
Comment (hjdivad) [inline lib/oasis/webworker_adapter.js]: probably should remove this

File: test/iframe_sandbox_tests.js [hunk 3]
```diff
@@ -279,3 +279,50 @@ test("HTML Sandboxes do not verify the Oasis URL", function() {
 
   sandbox.start();
 });
+
+test("JS Sandboxes can not load resources from a different domain than the bootstrap", function() {
+  expect(1);
+  stop();
+
+  var externalDomain = location.protocol + "//" + location.hostname + ":" + (parseInt(location.port, 10) + 3);
```
Comment (hjdivad) [inline test/iframe_sandbox_tests.js]: different origin, not a different domain

**Pull request merged**